jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -25,46 +89,76 @@ protected Action(Tween... tweens) {
         }
     }
 
+    /**
+     * Gets the length aka the duration of the current whole action that encloses Tweens.
+     * @return the length of the action in seconds.
+     */
     @Override
     public double getLength() {
         return length;
     }
 
     /**
-     * Alter the length (duration) of this Action.  This can be used to extend
-     * or truncate an Action.
-     *
+     * Alter the length (duration) of this Action.
+     * This can be used to extend or truncate an Action.
      * @param length the desired length (in unscaled seconds, default=0)
      */
     public void setLength(double length) {
         this.length = length;
     }
 
+    /**
+     * Gets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each Layer#advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * Default speed is 1.0, it plays the animation clips at their normal speed.
+     * @return the speed of frames.
+     */
     public double getSpeed() {
         return speed;
     }
 
+    /**
+     * Sets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each Layer#advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * Default speed is 1.0, it plays the animation clips at their normal speed.
+     * @param speed the speed of frames.
+     */
     public void setSpeed(double speed) {
         this.speed = speed;
-        if( speed < 0){
-            setForward(false);
-        } else {
-            setForward(true);
-        }
+        setForward(speed < 0);
     }
 
+    /**
+     * Gets the animation mask for this action.
+     * the animation mask holds data (subsets of an object) about which part of the animation would be played.
+     * @return the animation mask instance.
+     */
     public AnimationMask getMask() {
         return mask;
     }
 
+    /**
+     * Sets the animation mask for this action.
+     * the animation mask holds the info about which part (subsets of an object) of the animation to play, such as joints in {@link com.jme3.anim.ArmatureMask#addFromJoint(Armature, String)}.
+     * @param mask the animation mask instance.
+     */
     public void setMask(AnimationMask mask) {
         this.mask = mask;
     }
 
+    /**
+     * Tests the value of the forward flag.
+     * @return true if the animation action is in a forward manner state.
```

Comment:
It's not clear what a "forward manner state" is, nor what `false` indicates. I believe this method tests whether the action is running forward (true) or backward (false).

Reply:
That's true indeed.